components/formats-gpl/src/loci/formats/in/SlideBook6Reader.java

```diff
@@ -0,0 +1,402 @@
+			int numSeries = getNumCaptures();
+
+			core.clear();
+			for (int i=0; i<numSeries; i++) {
+				CoreMetadata ms = new CoreMetadata();
+				core.add(ms);
+				ms.sizeX = getNumXColumns(i);
+				if (ms.sizeX % 2 != 0) ms.sizeX++;
+				ms.sizeY = getNumYRows(i);
+				ms.sizeZ = getNumZPlanes(i);
+				ms.sizeT = getNumTimepoints(i);
+				ms.sizeC = getNumChannels(i);
+				int bytes = getBytesPerPixel(i);
+				if (bytes % 3 == 0) {
+					ms.sizeC *= 3;
+					bytes /= 3;
+					ms.rgb = true;
+				}
+				else ms.rgb = false;
+
+				ms.pixelType = FormatTools.pixelTypeFromBytes(bytes, false, true);
+				ms.imageCount = ms.sizeZ * ms.sizeT;
+				if (!ms.rgb) ms.imageCount *= ms.sizeC;
+				ms.interleaved = true;
+				ms.littleEndian = true;
+				ms.dimensionOrder = "XYCZT";
+				ms.indexed = false;
+				ms.falseColor = false;
+			}
+		}
+		catch (UnsatisfiedLinkError e) {
+			throw new MissingLibraryException(GENERAL_3I_MSG, e);
+		}
+		catch (Exception e) {
+			throw new MissingLibraryException(GENERAL_3I_MSG, e);
+		}
+
+		// fill in meta data
+		MetadataStore store = makeFilterMetadata();
+		MetadataTools.populatePixels(store, this);
+		int numCaptures = getNumCaptures();
+		if (getMetadataOptions().getMetadataLevel() != MetadataLevel.MINIMUM) {
+			for (int capture=0; capture<numCaptures; capture++) {
+				String imageName = getImageName(capture);
+				store.setImageName(imageName, capture);
+				String imageDescription = getImageComments(capture);
+				store.setImageDescription(imageDescription, capture);
+			}
+		}
+
+		// link Instrument and Image
+		String instrumentID = MetadataTools.createLSID("Instrument", 0);
+		store.setInstrumentID(instrumentID, 0);
+		for (int capture=0; capture < numCaptures; capture++) {
+			store.setImageInstrumentRef(instrumentID, capture);
+		}
+
+		// set voxel size per image (microns)
+		for (int capture=0; capture < numCaptures; capture++) {
+			double voxelsize = getVoxelSize(capture);
+			Length physicalSizeX = FormatTools.getPhysicalSizeX(voxelsize);
+			Length physicalSizeY = FormatTools.getPhysicalSizeY(voxelsize);
+			if (physicalSizeX != null) {
+				store.setPixelsPhysicalSizeX(physicalSizeX, capture);
+			}
+			if (physicalSizeY != null) {
+				store.setPixelsPhysicalSizeY(physicalSizeY, capture);
+			}
+			double stepSize = 0;
+			if (getNumZPlanes(capture) > 1) {
+				double plane0 = getZPosition(capture, 0, 0);
+				double plane1 = getZPosition(capture, 0, getNumChannels(capture));
+				// distance between plane 0 and 1 is step size, assume constant for all planes
+				stepSize = plane1 - plane0;
+			}
+
+			Length physicalSizeZ = FormatTools.getPhysicalSizeZ(stepSize);
+			if (physicalSizeZ != null) {
+				store.setPixelsPhysicalSizeZ(physicalSizeZ, capture);
+			}
+
+			// link other meta data to images
+			int numPositions = getNumPositions(capture);
+			int numTimepoints = getNumTimepoints(capture);
+			int numChannels = getNumChannels(capture);
+			int numZPlanes = getNumZPlanes(capture);
```

Comment:
It might be a little faster to use the ```sizeZ```, ```sizeT```, ```sizeC``` values from CoreMetadata (as set above) instead of making additional native method calls here.